controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/AbstractRequestProcessor.java

```diff
@@ -66,8 +67,20 @@
 @Slf4j
 public abstract class AbstractRequestProcessor<T extends ControllerEvent> extends SerializedRequestHandler<T> 
         implements StreamRequestProcessor {
-    protected static final Predicate<Throwable> OPERATION_NOT_ALLOWED_PREDICATE = e -> Exceptions.unwrap(e) 
-            instanceof StoreException.OperationNotAllowedException;
+    protected static final Predicate<Throwable> ILLEGAL_STATE_PREDICATE = e -> Exceptions.unwrap(e) instanceof StoreException.IllegalStateException;
+    protected static final Predicate<Throwable> DATA_NOT_FOUND_PREDICATE = e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException;
+    protected static final Predicate<Throwable> SEGMENT_NOT_FOUND_PREDICATE = e -> Exceptions.unwrap(e) instanceof StoreException.DataContainerNotFoundException;
+    protected static final Predicate<Throwable> EVENT_RETRY_PREDICATE = ILLEGAL_STATE_PREDICATE.or(DATA_NOT_FOUND_PREDICATE)
+                                                                            .or(SEGMENT_NOT_FOUND_PREDICATE)
+                                                                            .or(e -> Exceptions.unwrap(e) instanceof IllegalArgumentException)
+                                                                            .or(e -> Exceptions.unwrap(e) instanceof NullPointerException)
+                                                                            .negate();
+    protected static final Predicate<Throwable> SCALE_EVENT_RETRY_PREDICATE = EVENT_RETRY_PREDICATE
+                                                                                .or(e -> e instanceof EpochTransitionOperationExceptions.ConflictException)
+                                                                                .or(e -> e instanceof EpochTransitionOperationExceptions.ConditionInvalidException)
```

Comment:
this seems incorrect.. we want to retry on conflict exception. 

if you look at existing condition it is using conflict as a retry predicate condition. 
so we should remove it from here because now we are negating.

Reply:
also does the negate at the end also negate the event retry predicate? that would be incorrect too

Reply:
Fixed this.